waiter/src/waiter/scheduler/kubernetes.clj

```diff
@@ -596,7 +596,29 @@
   [{:keys [service-id->service-description-fn]} service-id]
   (service-id->service-description-fn service-id))
 
-(defn- get-services
+(defn- prettify-quota-error
+  "If the given message matches an expected format for a quota limitation event, transform it to
+   a more user-friendly message. Otherwise, return the original message."
+  [raw-message namespace]
+  (let [quota-regex #".*Error creating: pods \"([^\"]+)\" is forbidden: exceeded quota: pods, (.*)"
+        matches (re-find quota-regex raw-message)]
+    (if matches
+      (str "Could not create pod (exceeded quota in namespace " namespace ") - " (last matches))
+      raw-message)))
+
+(defn- get-k8s-event-deployment-error
+  "Extract a deployment error if the given service includes a k8s event that is preventing
+   deployment."
+  [{:keys [instances k8s/events k8s/namespace task-stats]}]
+  (let [{:keys [message reason]} (last events)
+        running-count (:running task-stats)]
+    (when (and (number? instances)
+               (pos? instances)
+               (= 0 running-count)
+               (= reason "FailedCreate"))
+      (prettify-quota-error message namespace))))
```

Comment:
How do we know for sure that it is a quota error message at this point? Should the method name be more generic, `prettify-failed-create-error`?